presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/optimization/JdbcComputePushdown.java

```diff
@@ -105,7 +112,12 @@ public PlanNode visitFilter(FilterNode node, Void context)
                 return node;
             }
 
-            // TODO: FilterRowExpression is currently mocked, needs to be implemented
+            DefaultRowExpressionTranslator<String> rowExpressionTranslator = new DefaultRowExpressionTranslator<>(translationContext);
+            TargetExpression<String> translatedFilter = defaultTranslate(
+                    rowExpressionTranslator,
+                    node.getPredicate(),
+                    new DefaultRowExpressionTranslator.ConnectorContext(oldTableScanNode.getAssignments()));
+            // TODO: put the translated filter into JdbcTableLayoutHandle
```

Comment:
Let's implement this and add the corresponding logic in `com.facebook.presto.plugin.jdbc.QueryBuilder`.